### DISCUSSION TITLE
Compressed binary

### DISCUSSION THREAD

**Discussion by drhideg** [benzino77/tasmocompiler#107] [Ideas]
I would like to make a download button for the compressed binary, which is made by gzip-firmware.py in the build_output/firmware folder. A little ugliness in it is that I don't understand why firmware.bin.bin is copied by name-firmware.py and it is compressed in the gz. How can i avoid that behavior?

**Comment by benzino77** [comment]
You want to avoid copying or compressing the file? Or both?

**Comment by benzino77** [comment]
There is something strange. Compiling Tasmota by hand it produces `tasmota.bin` and `tasmota.bin.gz` files in build_output/firmware, compilation by TasmoCompiler prodeuce `firmware.bin.bin` and `firmware.bin.bin.gz` files (double bin extension). 
Probably that's what you ware asking for ;)

**Comment by benzino77** [comment]
The problem is this line:
https://github.com/benzino77/tasmocompiler/blob/d693d040b034535eb4b82745376273c167a66a98/server/compile/platformio.ini#L4

this will produce double `bin.bin` extension. I do believe that it is a problem with platformio itself.

**Comment by drhideg** [comment]
I answer it in batch :)
I try to avoid including new dependecies and not necessary copy or compress on any nothing like happens before i came here.
If i remove  build_dir = .pioenvs everything seems to be fine except firmware.bin got the name from env:custom as custom.bin. 
To stay on the filname firmware.bin i must rename env to firmware.
Is this acceptable by you?
https://github.com/benzino77/tasmocompiler/compare/master...drhideg:drhideg-patch-compressed

**Comment by benzino77** [comment]
Looks promising. I will have more time next week to look at it a little bit closer. 
Meanwhile please check if your changes works fine for version 8.2.0 and up (as I remember this is a start version when gziped firmware can be uploaded).

**Comment by drhideg** [comment]
You are absolutely right! I did not check that.

I have not found a proper way to check gzipped binary with fs.existsSync with tasmotaRepo variable, or get tasmotaVersion without implement (copy) getTasmotaVersion function.
I always ran into restrict import from src/ folder error.
Now i check the compiler output contains 'bin_gzip' string (which I don’t feel is a professional solution) to determine compressed binary is made or not.
From Tasmota 8.2.0-8.4.0 the compressed binary in build_output is named build.bin.gz.

Also i found they drop support for older cores which lead to a failed compile.

**Comment by benzino77** [comment]
I think it's time to drop support (remove from the list of available versions one can choose) for Tasmota versions prior 8.5.0.
The same with old core versions (older than 2.7.x ??). 

It will solve the problem with the `<env>.bin` and `build.bin` file names mismatch and failures with compiling Tasmota with prehistoric Arduino Core versions.

**Comment by drhideg** [comment]
It must be you decision to drop support or not. I think userbase of this tool always use the latest version+core, the others can compile themselves.
I found that they drop support for older core than 2.7.1 happened in v8.4.0.

**Comment by benzino77** [comment]
It will be pull request from your side - so there is time to review and discuss changes. Having ability to download gzipped binary is tempting, so please prepare changes with dropped support for things we've discussed above. I also believe that users mostly use TC to compile development or latest stable with latest Arduino Core.

**Comment by benzino77** [comment]
My Internet is still not working but I can use my mobile as AP.
Your PR works as expected - good job. But there is a problem with version 8.5.0 of Tasmota. It produce `build.bin` and `build.bin.gz` instead `firmware.bin` and `firmware.bin.gz` (or in general `<platformio env>.bin`).

**Comment by drhideg** [comment]
Please don't apologize, it happens sometimes.
Weird :/
I fired up a fresh gitpod from the PR's [branch ](https://gitpod.io/#https://github.com/drhideg/tasmocompiler/tree/drhideg-patch-compressed). `build.bin` and `build.bin.gz`  only made with v8.2.0, v8.3.0, v8.3.1, v8.4.0 which will be removed from TasmoCompiler.

**Comment by benzino77** [comment]
That was something on my side. I have re-downloaded Tasmota repository and indeed v8.5.0 works as expected.
Great work! Thanks!
